test/ERC1155Tradable.js

```diff
@@ -0,0 +1,390 @@
+/* libraries used */
+
+const truffleAssert = require('truffle-assertions');
+
+
+/* Contracts in this test */
+
+const ERC1155Tradable = artifacts.require("../contracts/ERC1155Tradable.sol");
+const MockProxyRegistry = artifacts.require(
+  "../contracts/MockProxyRegistry.sol"
+);
+
+
+/* Useful aliases */
+
+const toBN = web3.utils.toBN;
+
+
+contract("ERC1155Tradable - ERC 1155", (accounts) => {
+  const NAME = 'ERC-1155 Test Contract';
+  const SYMBOL = 'ERC1155Test';
+  const URI_ROOT = 'https://a.b.c/def/hji/';
+  const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
+
+  const INITIAL_TOKEN_ID = 1;
+  const NON_EXISTENT_TOKEN_ID = 99999999;
+  const INITIAL_TOKEN_SUPPLY = 2;
+  const MINT_AMOUNT = 2;
+
+  const OVERFLOW_NUMBER = toBN(2, 10).pow(toBN(256, 10)).sub(toBN(1, 10));
+
+  const owner = accounts[0];
+  const creator = accounts[1];
+  const userA = accounts[2];
+  const userB = accounts[3];
+  const proxyForOwner = accounts[5];
+
+  let instance;
+  let proxy;
+
+  // Keep track of token ids as we progress through the tests, rather than
```

Comment:
I added a comment to unpack this:

  // For example if test A assumes that it will create token ID 1 and test B
  // assumes that it will create token 2, changing test A later so that it
  // creates another token will break this as test B will now create token ID 3.
  // Doing this avoids this scenario.